File: 74-search-a-2d-matrix/74-search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        
        row_size=len(matrix)
        
        for i in range(row_size):
            
            if matrix[i].count(target)>=1:
                
                return True
            
        return False

File: 74-search-a-2d-matrix/74-search-a-2d-matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        
        int row_size=matrix.size();
        int col_size=matrix[0].size();
        
        int curr_row = 0, curr_col = col_size - 1;
        
        while(curr_row < row_size && curr_col > -1){
            
            if(matrix[curr_row][curr_col] == target)
                return true;
            
            if(target > matrix[curr_row][curr_col])
                curr_row++;
            else
                curr_col--;
            
        }
        
        return false;
    }
};